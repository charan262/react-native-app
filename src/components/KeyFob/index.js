/**
 * @flow
 */

import React, { Component } from 'react';
import { Animated, StyleSheet, TouchableHighlight, View } from 'react-native';
import { connect } from "react-redux";
import { notifyKeyFobPress } from "../../actions/keyFob";
import ActivityOverlay from "../ActivityOverlay";

type Props = {};
const lock = "LOCK";
const unlock = "UNLOCK";

class KeyFob extends Component<Props> {
  state = {
    rotateValue: new Animated.Value(0),
    keyFobPressDisabled: false,
  };

  constructor() {
    super();
    this.state.spinAnimation = Animated.loop(
      Animated.timing(
        this.state.rotateValue,
        {
          toValue: 1,
          duration: 800
        }
      )
    );

    this.state.spin = this.state.rotateValue.interpolate({
      inputRange: [0, 1],
      outputRange: ['0deg', '360deg']
    });
  }

  async keyFobPress(type) {
    this.toggleKeyFobDisabled();
    this.state.spinAnimation.start();
    await this.props.notifyKeyFobPress(type);
    this.toggleKeyFobDisabled();
    this.state.spinAnimation.stop();
    this.state.spinAnimation.reset();
  }

  toggleKeyFobDisabled() {
    const disabled = !this.state.keyFobPressDisabled;
    this.setState(() => ({
      keyFobPressDisabled: disabled
    }))
  }

  render() {
    let { keyFobPressDisabled, spin } = this.state;

    return (
      this.props.keyFobInProgress ?
        <ActivityOverlay condition={this.props.keyFobInProgress} /> :
        <View style={styles.container}>
          <TouchableHighlight
            style={styles.button}
            onPress={() => {
              this.keyFobPress(lock);
            }}
            disabled={keyFobPressDisabled}
            underlayColor={'#b9ccee'}>
            <Animated.Image
              style={[styles.image, { transform: [{ rotate: spin }] }]}
              source={require('./img/lock.png')} />
          </TouchableHighlight>
          <View style={{ height: 88 }} />
          <TouchableHighlight
            style={styles.button}
            onPress={() => {
              this.keyFobPress(unlock);
            }}
            disabled={keyFobPressDisabled}
            underlayColor={'#b9ccee'}>
            <Animated.Image
              style={[styles.image, { transform: [{ rotate: spin }] }]}
              source={require('./img/unlock.png')} />
          </TouchableHighlight>
        </View>
    );
  }
}

const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    backgroundColor: '#fafafa',
    borderWidth: 4,
    borderColor: '#27467f',
    borderRadius: 100,
    height: 100,
    justifyContent: 'center',
    padding: 8,
    width: 100,
  },
  container: {
    alignItems: 'center',
    backgroundColor: '#f5Fcff',
    flex: 1,
    justifyContent: 'center',
  },
  image: {
    tintColor: '#5e81bc',
  }
});

const mapStateToProps = state => (
  {
    keyFob: state.keyFob,
    keyFobFailed: state.keyFobFailed,
    keyFobInProgress: state.keyFobInProgress,
  }
);

const mapDispatchToProps = dispatch => (
  {
    notifyKeyFobPress: toggleType => dispatch(notifyKeyFobPress(toggleType)),
  }
);

export default connect(mapStateToProps, mapDispatchToProps)(KeyFob);