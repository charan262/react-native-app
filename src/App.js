/**
 * @flow
 */

import React, { Component } from 'react';
import { AsyncStorage, Text } from "react-native";
import { DrawerNavigator, StackNavigator, SwitchNavigator } from "react-navigation";
import { Provider } from "react-redux";
import store from "./store";
import { authSuccessful } from "./actions/auth";
import { KeyFob, Login, Manual } from "./components";

type Props = {};

const Navigator = SwitchNavigator({
  Login: { screen: Login },
  Main: {
    screen: StackNavigator({
      Drawer: {
        screen: DrawerNavigator({
          Home: { screen: KeyFob },
          Manual: { screen: Manual },
        })
      }
    }, {
      navigationOptions: ({ navigation }) => ({
        headerLeft: <Text
          onPress={() => navigation.navigate('DrawerToggle')}
          style={{
            marginLeft: 10,
            fontWeight: 'bold',
          }}>MENU</Text>,

        headerRight: <Text
          onPress={() => {
            store.dispatch(authSuccessful(''));
            navigation.navigate('Login')
          }}
          style={{
            marginRight: 10,
            fontWeight: 'bold',
          }}>LOGOUT</Text>
      })
    })
  }
});

AsyncStorage.getItem('@GM:JWT')
  .then(value => store.dispatch(authSuccessful(value)));

export default class App extends Component<Props> {
  render() {
    return (
      <Provider store={store}>
        <Navigator />
      </Provider>
    );
  }
}
