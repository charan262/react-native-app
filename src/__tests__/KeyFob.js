import 'react-native';
import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer';
import KeyFob from '../components/KeyFob';
// import Adapter from 'enzyme-adapter-react-15';
// import { shallow } from 'enzyme';

it('renders correctly', () => {
  const component = renderer.create(
    <KeyFob />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


// it('disables KeyFob when making async call', done => {
//     const wrapper = shallow(<KeyFob />);
//     wrapper.instance().keyFobPress("LOCK");
// });
