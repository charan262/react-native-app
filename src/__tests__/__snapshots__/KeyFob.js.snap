// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f5Fcff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fafafa",
        "borderColor": "#27467f",
        "borderRadius": 100,
        "borderWidth": 4,
        "height": 100,
        "justifyContent": "center",
        "padding": 8,
        "width": 100,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Image
      collapsable={undefined}
      source={Object {}}
      style={
        Object {
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "height": 88,
      }
    }
  />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fafafa",
        "borderColor": "#27467f",
        "borderRadius": 100,
        "borderWidth": 4,
        "height": 100,
        "justifyContent": "center",
        "padding": 8,
        "width": 100,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Image
      collapsable={undefined}
      source={Object {}}
      style={
        Object {
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
        }
      }
    />
  </View>
</View>
`;
